libraries/ESP8266WiFiMesh/src/CompatibilityLayer.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+  Old version of ESP8266WiFiMesh.cpp - Mesh network node
+  Sets up a Mesh Node which acts as a router, creating a Mesh Network with other nodes. All information
+  is passed in both directions, but it is up to the user what the data sent is and how it is dealt with.
+ 
+  Copyright (c) 2015 Julian Fell. All rights reserved.
+ 
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+
+
+
+
+
+/********************************************************************************************
+* NOTE!
+*
+* All method signatures in this file are deprecated and will be removed in core version 2.5.0.
+* If you are still using these methods, please consider migrating to the new API shown in 
+* the ESP8266WiFiMesh.h source file.
+*
+* TODO: delete this file.
+********************************************************************************************/
+
+
+
+
+
+#include <Arduino.h>
+#include <ESP8266WiFi.h>
+#include <WiFiClient.h> 
+#include <WiFiServer.h>
+
+#include "ESP8266WiFiMesh.h"
+
+#define SSID_PREFIX      		"Mesh_Node"
+#define SERVER_IP_ADDR			"192.168.4.1"
+#define SERVER_PORT				4011
+
+// DEPRECATED!
+ESP8266WiFiMesh::ESP8266WiFiMesh(uint32_t chip_id, std::function<String(String)> handler)
+: _server(SERVER_PORT)
+{
+	_chip_id = chip_id;
+	_ssid = String( String( SSID_PREFIX ) + String( _chip_id ) );
+	_ssid_prefix = String( SSID_PREFIX );
+	_handler = handler;
+}
+
+/**
+ * Wait for a WiFiClient to connect
+ *
+ * @returns: True if the client is ready, false otherwise.
+ * 
+ */
+// DEPRECATED!
+bool ESP8266WiFiMesh::waitForClient(WiFiClient curr_client, int max_wait)
+{
+	int wait = max_wait;
+	while(curr_client.connected() && !curr_client.available() && wait--)
+		delay(3);
+
+	/* Return false if the client isn't ready to communicate */
+	if (WiFi.status() == WL_DISCONNECTED || !curr_client.connected())
+		return false;
+	
+	return true;
+}
+
+/**
+ * Send the supplied message then read back the other node's response
+ * and pass that to the user-supplied handler.
+ *
+ * @target_ssid The name of the AP the other node has set up.
+ * @message The string to send to the node.
+ * @returns: True if the exchange was a succes, false otherwise.
+ * 
+ */
+// DEPRECATED!
+bool ESP8266WiFiMesh::exchangeInfo(String message, WiFiClient curr_client)
```

Comment:
Can message be declared reference without losing backwards compat?
Same for curr_client.